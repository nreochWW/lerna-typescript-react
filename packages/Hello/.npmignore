src/
dist/
legacy/
stories/
__tests__
tsconfig.json
webpack.config.js