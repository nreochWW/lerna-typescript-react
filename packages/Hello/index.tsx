import * as React from "react";

export interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <h1>
    Hello from {props.compiler} and {props.framework}!
  </h1>
);

export default Hello;
