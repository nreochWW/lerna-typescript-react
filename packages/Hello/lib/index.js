"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const world_typescript_react_1 = require("world-typescript-react");
const Hello = (props) => (React.createElement("div", { style: { border: "1px solid blue", padding: "10px" } },
    React.createElement("h1", null,
        "Hello World from ",
        props.compiler,
        " and ",
        props.framework,
        "!"),
    React.createElement(world_typescript_react_1.default, { compiler: "TypeScript", framework: "React" })));
exports.default = Hello;
