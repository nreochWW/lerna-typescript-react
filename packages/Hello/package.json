{
  "name": "hello-typescript-react",
  "version": "0.0.1",
  "main": "index.tsx",
  "license": "MIT"
}
