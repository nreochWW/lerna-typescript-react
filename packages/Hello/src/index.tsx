import * as React from "react";
import World from "world-typescript-react";

export interface HelloProps {
  compiler: string;
  framework: string;
}

const Hello = (props: HelloProps) => (
  <div style={{ border: "1px solid blue", padding: "10px" }}>
    <h1>
      Hello World from {props.compiler} and {props.framework}!
    </h1>
    <World compiler="TypeScript" framework="React" />
  </div>
);

export default Hello;
