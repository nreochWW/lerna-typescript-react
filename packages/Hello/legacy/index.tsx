import * as React from "react";
import * as ReactDOM from "react-dom";

import Hello from "../src/index";

ReactDOM.render(
  <Hello compiler="TypeScript" framework="React" />,
  document.getElementById("hello")
);
