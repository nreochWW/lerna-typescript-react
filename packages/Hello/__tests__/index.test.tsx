import * as React from "react";
import { shallow } from "enzyme";
import Hello from "../src/index";

describe("Hello package tests", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<Hello compiler="TypeScript" framework="React" />);
    expect(wrapper).toMatchSnapshot();
  });
});
