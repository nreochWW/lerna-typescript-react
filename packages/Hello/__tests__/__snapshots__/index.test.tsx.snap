// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello package tests matches snapshot 1`] = `
<div
  style={
    Object {
      "border": "1px solid blue",
      "padding": "10px",
    }
  }
>
  <h1>
    Hello World from 
    TypeScript
     and 
    React
    !
  </h1>
  <World
    compiler="TypeScript"
    framework="React"
  />
</div>
`;
